src/main/java/com/bina/varsim/types/variant/Variant.java

```diff
@@ -1136,15 +1136,45 @@ public void output(PrintWriter p) {
      * here it guarantees to return
      * genotype 1 or 2
      *
-     * @return
+     * when both maternal and paternal genotypes are
+     * unknown, i.e. -1, then we need to return
+     * something more meaningful than just 1. however
+     * this is by no means a very good return
+     * value. proper fix involves explicitly handling
+     * of missing genotype/phasing information.
+     *
+     * @return genotype number (0,1,2) or 1 for
+     * missing genotype (or maximum possible)
      */
     public byte getGoodPaternal() {
-        return (paternal < 0) ? 1 : paternal;
-
+        return (byte) ((paternal < 0) ? Math.min(1,alts.length) : paternal);
     }
 
+    /**
+     * if genotype missing and return types are iterated over
+     * 4 possible scenarios:
+     *
+     * X: 2 or max possible
+     * Y: 1 or max possible
+     * MT: 1 or max possible
+     * autosomal: 2 or max possible
+     *
+     * here we do not have gender information (sometimes
+     * it's unavailable), so the maternal haploid chromosome
+     * could be actually diploid chromosome, and the reverse
+     * is also possible. we just pretend it is diploid.
+     *
+     * @return exact genotype number (0,1,2) or most comprehensive
+     * genotype number
+     */
     public byte getGoodMaternal() {
-        return (maternal < 0) ? 1: maternal;
+        if (maternal >= 0)
+            return maternal; //not missing
+        //missing
+        if (chr.isY() || chr.isMT()) {
```

Comment:
Here it's really a dirty hack. I want to have `getGoodPaternal` and `getGoodMaternal` collectively return all possible alleles indices given no `GT`. Since there is always at least one ALT allele, `getGoodPaternal` will return 1 when GT is missing; while `getGoodMaternal` will return 2 most of the time when GT is missing. Exception is that when variant is on Y or MT, then at most 1 ALT can be present, so in both cases, 1 is returned. I really want to extend VarSim's capability to handle a population-based VCF, though.